### PR TITLE
[INTEG-976] Add ESLint to AICG and fix errors

### PR DESCRIPTION
**Pull request by whitelisab**
## Purpose

We did not have ESLint configured for the AI Content Generator app. This PR adds a standard configuration, and fixes the errors and warnings in the app.

## Approach

Installed ESLint as a dev dependency and and ran `npm init @eslint/config` to get a basic configuration set up using plugins for react, react hooks, and typescript. 

I temporarily added some files to the `.eslintignore`, namely the V1 components and test files, along with TODOs to clean those up later.

## Testing steps

Test the basic actions in the app to ensure that it all behaves as expected. Run `npm run lint` and you shouldn't see any warnings or errors.

### CODE REVIEW

File: apps/ai-content-generator/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "env": {
+    "browser": true,
+    "es2021": true
+  },
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:react/recommended",
+    "plugin:react/jsx-runtime",
```
Comment (whitelisab) [inline apps/ai-content-generator/.eslintrc.json:10]: This prevents the error `'React' must be in scope when using JSX`

File: apps/ai-content-generator/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "plugin:react-hooks/recommended"
+  ],
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaVersion": "latest",
+    "sourceType": "module"
+  },
+  "plugins": ["@typescript-eslint", "react"],
+  "rules": {
+    "react/no-unknown-property": ["error", { "ignore": ["css"] }]
```
Comment (whitelisab) [inline apps/ai-content-generator/.eslintrc.json:20]: This prevents the error `Unknown property 'css' found`, see more info here: https://emotion.sh/docs/eslint-plugin-react

**Review by ryunsong-contentful — APPROVED** [review]
Worked for me!

File: apps/ai-content-generator/src/hooks/config/useInitializeParameters.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ import { ParameterAction, ParameterReducer } from '@components/config/parameterR
 const useInitializeParameters = (dispatch: Dispatch<ParameterReducer>) => {
   const sdk = useSDK<ConfigAppSDK>();
 
-  const dispatchParameters = async () => {
+  const dispatchParameters = useCallback(async () => {
```
Comment (rusticpenguin) [inline apps/ai-content-generator/src/hooks/config/useInitializeParameters.ts:17]: Same comment as the `useSaveConfigHandler.ts`, does this resolve any potential re-rendering issue?

File: apps/ai-content-generator/src/hooks/config/useSaveConfigHandler.ts [hunk 4]
```diff
@@ -12,23 +12,23 @@ import { useEffect } from 'react';
...
     const currentState = await sdk.app.getCurrentState();
 
     return {
       parameters,
       targetState: currentState,
     };
-  };
+  }, [parameters, sdk.app]);
 
-  const changeSaveConfigHandler = () => {
```
[the remark this answers is not in the document]
Comment (whitelisab) [reply apps/ai-content-generator/src/hooks/config/useSaveConfigHandler.ts:24]: Using `useCallback` in this hook and the other one didn't resolve any re-rendering issues, but as we discussed there are some other improvements we can make on the Config page that we can address in another PR!

**Pull request merged**